filters/filters.go

```diff
@@ -117,7 +119,17 @@ func (r Registry) Register(s Spec) {
 }
 
 // Metrics provides an option to collect custom metrics
-type Metrics interface {
-	IncCounter(key string)
-	MeasureSince(key string, start time.Time)
+const MetricKey = "%s.%s"
+
+type FilterMetrics struct {
+	Metrics    *metrics.Metrics
+	FilterName string
```

Comment:
i would prefer not using this as a package visible field. Let me contemplate a bit on a different approach..